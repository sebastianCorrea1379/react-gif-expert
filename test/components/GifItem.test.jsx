import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem.jsx', () => {
    
    const title = 'Sebas';
    const url = 'https://sebas.com/sebas.jpg';

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });


    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });


    test('Debe mostrar el titulo en el componente', () => {
       
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
        
    });

});