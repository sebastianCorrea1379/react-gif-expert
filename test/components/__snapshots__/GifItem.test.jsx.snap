// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem.jsx Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Sebas"
      src="https://sebas.com/sebas.jpg"
    />
    <p>
      Sebas
    </p>
  </div>
</div>
`;
